source/AI.cpp

```diff
@@ -2786,7 +2800,31 @@ void AI::AimTurrets(const Ship &ship, Command &command, bool opportunistic) cons
 				p += v * rendezvousTime;
 				
 				// Determine how much the turret must turn to face that vector.
-				double degrees = (Angle(p) - aim).Degrees();
+				double degrees = 0.0;
+				Angle angleToPoint = Angle(p);
+				if(hardpoint.IsOmnidirectional())
+					degrees = (angleToPoint - aim).Degrees();
+				else
+				{
+					auto range = hardpoint.GetSweptAngle();
+					const Angle facing = ship.Facing();
+					range.first += facing;
+					range.second += facing;
```

Comment:
Seems like we should enable this automatically, i.e. pass the reference angle into the method, if this is a common usage of the method call. Otherwise the caller might forget to do this.
But, if this is a special scenario, then we should not.

(Right now this method is used 3 times, and 2 adjust the return value, while 1 does not.)

Reply:
I would not change it because any methods of Hardpoint except firing don't consider the direction of the ship.